daceml/transformation/reshape_elimination.py

```diff
@@ -0,0 +1,146 @@
+                print(
+                    "Automatically expanded library node \"{}\" with implementation \"{}\"."
+                    .format(str(node), impl_name))
+                # We made a copy of the original list of nodes, so we keep
+                # iterating even though this list has now changed
+                if recursive:
+                    expanded_something = True
+        if expanded_something:
+            states.append(state)  # Nodes have changed. Check state again
+
+
+@registry.autoregister_params(singlestate=True)
+@properties.make_properties
+class ReshapeElimination(xf.Transformation):
+    """ Merge a reshape into a preceding or following nested SDFG call.
+    """
+    # pattern matching only checks that the type of the node matches,
+    _reshape_node = xf.PatternNode(donnx.ONNXReshape)
+    _access_node = xf.PatternNode(nodes.AccessNode)
+    _nsdfg = xf.PatternNode(nodes.NestedSDFG)
+
+    @staticmethod
+    def expressions():
+        return [
+            sdfg_utils.node_path_graph(ReshapeElimination._reshape_node,
+                                       ReshapeElimination._access_node,
+                                       ReshapeElimination._nsdfg)
+        ]
+
+    @staticmethod
+    def can_be_applied(graph: dace.sdfg.graph.OrderedMultiDiConnectorGraph,
+                       candidate: Dict[nodes.Node, int],
+                       expr_index: int,
+                       sdfg,
+                       strict: bool = False):
+
+        graph: dace.SDFGState
+        reshape_node = graph.nodes()[candidate[
+            ReshapeElimination._reshape_node]]
+        access_node = graph.nodes()[candidate[ReshapeElimination._access_node]]
+
+        if not sdfg.arrays[access_node.data].transient:
+            return False
+
+        in_memlet = utils.in_edge_with_name(reshape_node, graph, "data").data
+
+        def is_memlet_contiguous(mm):
+            if (not isinstance(mm.subset, subsets.Range)
+                    or any([step != 1 for _, _, step in mm.subset])):
+                return False
+            return True
+
+        # check that the in memlets is contiguous (this check can be relaxed)
+        for mm in [in_memlet] + [e.data for e in graph.out_edges(access_node)]:
+            if not is_memlet_contiguous(mm):
+                return False
+
+        def _prod(sequence):
+            return functools.reduce(lambda a, b: a * b, sequence, 1)
+
+        # check that the in arrays are contiguous
+        def is_desc_contiguous(desc):
+            expected_strides = [
+                _prod(desc.shape[i + 1:]) for i in range(len(desc.shape))
+            ]
+            return all(es == s
+                       for es, s in zip(expected_strides, desc.strides))
+
+        for desc in [
+                sdfg.arrays[in_memlet.data], sdfg.arrays[access_node.data]
+        ]:
+            if not is_desc_contiguous(desc):
+                return False
```

Comment:
Wouldn't all you want to check here is "if subset.coord_at(subset.min_element()) == view_subset.coord_at(view_subset.min_element()" (i.e., same pointer)?
@orausch

Reply:
I thought the Nested SDFG entry only handled contiguous reshapes (i.e. no changes in strides). But this was written before view nodes existed, not sure if we even need this whole transformation anymore. I'd propose to just leave it and update if we hit new cases.

Reply:
A nested SDFG used to be a primitive version of View. You can use the data in any type or shape/strides there, no restriction.

Sure, let's keep it as is.